setup_hooks.sh

```diff
@@ -1,4 +1,7 @@
 #!/bin/sh
-chmod +x "$PWD"/ci/lint/pre-push
-ln -s "$PWD"/ci/lint/pre-push "$PWD"/.git/hooks/pre-push
-ln -s "$PWD"/ci/lint/prepare-commit-msg "$PWD"/.git/hooks/prepare-commit-msg
+set -exu
+
+builtin cd "$(dirname "${BASH_SOURCE:-$0}")"
```

Comment:
Use the `builtin` it can reduce unnecessary troubles and it will not create new thread. 

I considered use `$(git rev-parse --show-toplevel)` to get the root, finally not adopted due to if the `hook script` file's location changed, `setup_hooks.sh` need to be modified simultaneously.

Any thoughts?

Reply:
we prefer using absolute path. like bazel all uses absolute path or local path. 

> if the hook script file's location changed, setup_hooks.sh need to be modified simultaneously.

sorry but I do not quite understand this statement.. if you use `$(git rev-parse --show-toplevel)`, then `setup_hooks.sh` can be moved to anywhere in this repo without changing its content, right?

with `$(dirname "${BASH_SOURCE:-$0}")` on the other hand, `setup_hooks.sh` has to stay at the repo's root?

it is okay to keep the `ln -sf ../..` (so that it does not use the original `$PWD`).

Reply:
I mean if the `./ci/lint/` be moved, this script will not work. Is there any better solution to resolve `ln -sf ../..`?

Reply:
Keep `builtin cd "$(dirname "${BASH_SOURCE:-$0}")"` avoid stops `git rev-parse` from failing if we run this from the .git directory.